.github/workflows/docker-publish.yml

```diff
@@ -21,6 +21,34 @@ jobs:
         if: ${{ github.event.inputs.tag }}
         run: GITHUB_TAG=${{ github.event.inputs.tag }}; echo ::set-output name=tag::${GITHUB_TAG#grain-}
 
+      - name: Start AWS EC2 arm64 instance
+        id: instance
+        run: |
+          aws ec2 start-instances --instance-ids ${{ secrets.AWS_EC2_INSTANCE_ID }}
+          aws ec2 wait instance-running --instance-ids ${{ secrets.AWS_EC2_INSTANCE_ID }}
+          echo ::set-output name=ip::$(aws ec2 describe-instances --filters "Name=instance-id,Values=${{ secrets.AWS_EC2_INSTANCE_ID }}" --query 'Reservations[*].Instances[*].[PublicIpAddress]' --output text)
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
+
+      - name: Bring in SSH keys
+        uses: webfactory/ssh-agent@v0.5.4
```

Comment:
Please fork this action and use our fork. People can swap out tags out from underneath and hijack things (I also trust you audited it?)

Reply:
Ideally, the AWS instance would use the public key from grainbot instead of us importing their private key (we'd need to add a public key to grainbot), but I have no idea how hard that would be

Reply:
I made this keypair, though yeah it'd be nice if grainbot was able to have its own private key. From what I can tell GitHub doesn't currently support it.